src/utils/slugify.ts

```diff
@@ -0,0 +1,36 @@
+export function slugify(input: string): string {
+  // See src/utils/__test__/slugify.test.ts for examples
+
+  const result = removeDiacritics(input).replaceAll(' ', '-').toLowerCase();
+
+  const unhandledCharacters = result.match(/[^abcdefghijklmnopqrstuvwxyz0123456789_.-]/g);
+  if (unhandledCharacters) {
+    const sortedUniqueCharacters = Array.from(new Set(unhandledCharacters)).sort();
+    throw new UnhandledCharactersError(sortedUniqueCharacters);
+  }
+
+  return result;
+}
+
+function removeDiacritics(input: string): string {
+  /*
+   Normalization form decomposition (NFD) splits characters like into their
```

Comment:
NIT: block comments generally have a * per line and will sometimes cause formatters to give weird outputs without them